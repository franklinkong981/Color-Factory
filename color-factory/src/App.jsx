import React, {useState} from 'react';
import {Routes, Navigate, Route} from "react-router-dom";

import ColorList from "./ColorList.jsx";
import NewColorForm from "./NewColorForm.jsx";
import FilterColors from "./FilterColors.jsx";

import './App.css';

const initialColors = [
  {name: "Red", hex: "#FF0000"},
  {name: "Yellow", hex: "#FFFF00"},
  {name: "Green", hex: "#00FF00"}
];

function App() {
  const [colors, setColors] = useState(initialColors);

  const addNewColor = (newColor) => {
    setColors([newColor, ...colors]);
  }

  const resetColors = () => {
    setColors(initialColors);
  };

  return (
    <div className="App">
      <Routes>
        <Route path="/colors" element={<ColorList colors={colors} resetColors={resetColors}/>} />
        <Route path="/colors/new" element={<NewColorForm addNewColor={addNewColor}/>} />
        <Route path="/colors/:color" element={<FilterColors colorsToFilter={colors} />} />
        <Route path="*" element={<Navigate to="/colors" />} />
      </Routes>
    </div>
  );
}

export default App;
