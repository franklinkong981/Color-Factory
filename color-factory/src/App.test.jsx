import React from "react";
import {render, fireEvent} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";

import App from "./App.jsx";

describe("Test <App/> rendering", function () {
  it("renders without crashing", function() {
    render((
      <MemoryRouter>
        <App/>
      </MemoryRouter>
    ));
  });
  it ("matches the snapshot for the App component", function() {
    const {asFragment} = render((
      <MemoryRouter>
        <App/>
      </MemoryRouter>
    ));
    expect(asFragment()).toMatchSnapshot();
  });
});

