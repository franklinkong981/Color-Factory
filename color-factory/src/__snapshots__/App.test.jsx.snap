// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test <App/> rendering > matches the snapshot for the App component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorList"
    >
      <h1
        class="ColorList-welcome"
      >
        Welcome to the Color Factory!
      </h1>
      <h2
        class="ColorList-select"
      >
        Please select a color below to view the color:
      </h2>
      <ul
        class="ColorList-list"
      >
        <li>
          <a
            class="ColorList-color"
            data-discover="true"
            href="/colors/Red"
          >
            Red
          </a>
        </li>
        <li>
          <a
            class="ColorList-color"
            data-discover="true"
            href="/colors/Yellow"
          >
            Yellow
          </a>
        </li>
        <li>
          <a
            class="ColorList-color"
            data-discover="true"
            href="/colors/Green"
          >
            Green
          </a>
        </li>
      </ul>
      <h2
        class="ColorList-new-color-header"
      >
        Or add a new color to the list
      </h2>
      <button
        class="ColorList-new-color-button"
      >
        <a
          data-discover="true"
          href="/colors/new"
        >
          Add New Color
        </a>
      </button>
      <button
        class="ColorList-rest-colors-button"
      >
        Reset Colors
      </button>
    </div>
  </div>
</DocumentFragment>
`;
